tests/robustness/validate/patch_history.go

```diff
@@ -54,7 +68,14 @@ func putRevision(reports []report.ClientReport) map[keyValue]int64 {
 	return requestRevision
 }
 
-func patchOperations(operations []porcupine.Operation, watchRevision, putReturnTime, clientPutCount, persistedPutCount map[keyValue]int64) []porcupine.Operation {
+func patchOperations(
+	operations []porcupine.Operation,
+	watchRevision, putReturnTime, clientPutCount, persistedPutCount map[keyValue]int64,
+	clientDeleteCount, persistedDeleteCount map[string]int64,
+	clientCompactCount, persistedCompactCount map[int64]int64,
+	deleteReturnTime map[string]int64,
+	compactReturnTime map[int64]int64,
+) []porcupine.Operation {
```

Comment:
There is a lot of arguments with that and easy to mix up. Have you considered separating Put and Delete by type?

For example instead of
```
putReturnTime, clientPutCount, persistedPutCount map[keyValue]int64,
deleteReturnTime, clientDeleteCount, persistedDeleteCount map[string]int64, 
```

Have
```
putReturnTime, clientPutCount, persistedPutCount map[PutOptions]int64,
deleteReturnTime, clientDeleteCount, persistedDeleteCount map[DeleteOptions]int64, 
```

And instead of 
```
putReturnTime, clientPutCount, persistedPutCount map[PutOptions]int64,
```
Have
```
type patchArgs struct{
  returnTime int64
  clientCount int64
  persistedCount int64
}

put map[PutOptions]patchArgs
```